__tests__/List.ts

```diff
@@ -484,6 +484,16 @@ describe('List', () => {
     expect(r.toArray()).toEqual(['b', 'd', 'f']);
   });
 
+  it('filters values based on type', () => {
+    class A {}
+    class B extends A {}
+    class C extends A {}
```

Comment:
Please make sure to add some methods to classes `B` and `C`. Because of structural typing, TS will be fine with this case, but if you do something like:

```ts
class A {}
class B extends A {
  public foo(): string {
    return "B";
  }
}
class C extends A {
  public bar(): string {
    return "C";
  }
}
```

then `let l2` will have a type error.